client/src/rubin/nublado/client/exceptions.py

```diff
@@ -0,0 +1,456 @@
+"""Exceptions for rsp_jupyter_client."""
+
+from __future__ import annotations
+
+import datetime
+import re
+from typing import Self
+
+import httpx
+from safir.datetime import format_datetime_for_logging
+from safir.slack.blockkit import (
+    SlackBaseBlock,
+    SlackBaseField,
+    SlackCodeBlock,
+    SlackException,
+    SlackMessage,
+    SlackTextBlock,
+    SlackTextField,
+    SlackWebException,
+)
+from websockets.exceptions import InvalidStatus, WebSocketException
+
+_ANSI_REGEX = re.compile(r"(?:\x1B[@-_]|[\x80-\x9F])[0-?]*[ -/]*[@-~]")
+"""Regex that matches ANSI escape sequences."""
+
+__all__ = [
+    "CodeExecutionError",
+    "JupyterProtocolError",
+    "JupyterTimeoutError",
+    "JupyterWebError",
+    "JupyterWebSocketError",
+    "MobuSlackException",
+    "MobuSlackWebException",
+]
+
+
+def _remove_ansi_escapes(string: str) -> str:
+    """Remove ANSI escape sequences from a string.
+
+    Jupyter labs like to format error messages with lots of ANSI escape
+    sequences, and Slack doesn't like that in messages (nor do humans want to
+    see them). Strip them out.
+
+    Based on `this StackOverflow answer
+    <https://stackoverflow.com/questions/14693701/>`__.
+
+    Parameters
+    ----------
+    string
+        String to strip ANSI escapes from.
+
+    Returns
+    -------
+    str
+        Sanitized string.
+    """
+    return _ANSI_REGEX.sub("", string)
+
+
+class MobuSlackException(SlackException):
```

Comment:
Will it be confusing to have the "Mobu" naming in the Nublado client?

Reply:
It will.  That was a transplantation oversight.

Reply:
Fixed.